rust/ddlog_benches/benches/citations.rs

```diff
@@ -0,0 +1,82 @@
+use criterion::{
+    black_box, criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion, Throughput,
+};
+use ddlog_benches::citations;
+use std::{ops::RangeInclusive, time::Duration};
+
+/// Benchmark all targets using 1, 2, 3, and 4 threads
+const DDLOG_WORKERS: RangeInclusive<usize> = 1..=4;
+
+// Increment the number of records taken by 100,000 until we're taking 1 million records
```

Comment:
Looks like there's just 1 increment happening here.  Does it make sense to go 100,000 in increments of 20,000? Actually, I'm not sure I see a lot of value in benchmarking loads of different dataset sizes. 2 or 3 is probably enough.

Reply:
So, this is my current conundrum: Criterion is amazing and gives statistically correct results, but even at the minimum sample size of 10 benchmarks still take upwards of 5 minutes *each*. The alternative would be some sort of "macro-benchmarking" rig that just runs each case 1 or 2 times and averages them, but we lose the sense of security that criterion provides. An alternative to both of these would be a hybrid approach, running smaller datasets under criterion while larger go under the macro rig, thoughts?

Reply:
It sounds like we want to stick to criterion.  Can the test suite be controlled from the command line? E.g., can we set things up so that on every commit we run a scaled down version (say, 5 iterations of shorter 2-minute tests) in CI, and then we can run longer, more reliable, benchmarks on demand, e.g., on a dedicated server or just locally on our machines?.

Reply:
I'll look into it, I think I should be able to make things controlled via cli so that we can scale things along to where they're needed (be it large & long benchmarks locally or small ones on CI)